MariaDB4jSampleTutorialTest.testLocalMariaDB fails when running e.g. on GitHub Codespace

```
[ERROR] Tests run: 3, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 7.679 s <<< FAILURE! - in ch.vorburger.mariadb4j.tests.MariaDB4jSampleTutorialTest
[ERROR] ch.vorburger.mariadb4j.tests.MariaDB4jSampleTutorialTest.testLocalMariaDB  Time elapsed: 0.122 s  <<< ERROR!
ch.vorburger.exec.ManagedProcessException: An error occurred while installing the database
        at ch.vorburger.mariadb4j.DB.install(DB.java:141)
        at ch.vorburger.mariadb4j.DB.newEmbeddedDB(DB.java:89)
        at ch.vorburger.mariadb4j.tests.MariaDB4jSampleTutorialTest.check(MariaDB4jSampleTutorialTest.java:70)
        at ch.vorburger.mariadb4j.tests.MariaDB4jSampleTutorialTest.testLocalMariaDB(MariaDB4jSampleTutorialTest.java:57)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
        at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
        at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
        at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:377)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:284)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:248)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:167)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
        at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
        at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
Caused by: ch.vorburger.exec.ManagedProcessException: Program [/usr/scripts/mysql_install_db, --datadir=/tmp/MariaDB4j/data/34685, --tmpdir=/tmp/MariaDB4j/tmp/34685, --basedir=/usr, --no-defaults, --force, --skip-name-resolve] (in working directory /usr) failed, exitValue=-559038737, last 100 lines of console:

        at ch.vorburger.exec.ManagedProcess.checkResult(ManagedProcess.java:310)
        at ch.vorburger.exec.ManagedProcess.startExecute(ManagedProcess.java:216)
        at ch.vorburger.exec.ManagedProcess.start(ManagedProcess.java:155)
        at ch.vorburger.mariadb4j.DB.install(DB.java:138)
        ... 31 more
Caused by: org.apache.commons.exec.ExecuteException: Execution failed (Exit value: -559038737. Caused by java.io.IOException: Cannot run program "/usr/scripts/mysql_install_db" (in directory "/usr"): error=2, No such file or directory)
        at org.apache.commons.exec.DefaultExecutor$1.run(DefaultExecutor.java:205)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.io.IOException: Cannot run program "/usr/scripts/mysql_install_db" (in directory "/usr"): error=2, No such file or directory
        at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1128)
        at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1071)
        at java.base/java.lang.Runtime.exec(Runtime.java:592)
        at org.apache.commons.exec.launcher.Java13CommandLauncher.exec(Java13CommandLauncher.java:61)
        at org.apache.commons.exec.DefaultExecutor.launch(DefaultExecutor.java:279)
        at org.apache.commons.exec.DefaultExecutor.executeInternal(DefaultExecutor.java:336)
        at org.apache.commons.exec.DefaultExecutor.access$200(DefaultExecutor.java:48)
        at org.apache.commons.exec.DefaultExecutor$1.run(DefaultExecutor.java:200)
        ... 1 more
Caused by: java.io.IOException: error=2, No such file or directory
        at java.base/java.lang.ProcessImpl.forkAndExec(Native Method)
        at java.base/java.lang.ProcessImpl.<init>(ProcessImpl.java:340)
        at java.base/java.lang.ProcessImpl.start(ProcessImpl.java:271)
        at java.base/java.lang.ProcessBuilder.start(ProcessBuilder.java:1107)
        ... 8 more

[INFO] Running ch.vorburger.mariadb4j.tests.springframework.MariaDB4jSpringServiceOverrideBySetTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 6.4 s - in ch.vorburger.mariadb4j.tests.springframework.MariaDB4jSpringServiceOverrideBySetTest
[INFO] Running ch.vorburger.mariadb4j.tests.springframework.MariaDB4jSpringServiceNewDefaultsOverridenBySpringValueTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.342 s - in ch.vorburger.mariadb4j.tests.springframework.MariaDB4jSpringServiceNewDefaultsOverridenBySpringValueTest
[INFO] Running ch.vorburger.mariadb4j.tests.springframework.MariaDB4jSpringServiceOverrideBySpringValueTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 6.839 s - in ch.vorburger.mariadb4j.tests.springframework.MariaDB4jSpringServiceOverrideBySpringValueTest
[INFO] Running ch.vorburger.mariadb4j.tests.springframework.MariaDB4jSpringServiceStandardDefaultsTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.63 s - in ch.vorburger.mariadb4j.tests.springframework.MariaDB4jSpringServiceStandardDefaultsTest
[INFO] Running ch.vorburger.mariadb4j.tests.MariaDB4jServiceTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 4.03 s - in ch.vorburger.mariadb4j.tests.MariaDB4jServiceTest
[INFO] Running ch.vorburger.mariadb4j.tests.MariaDB4jSampleOtherTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 21.91 s - in ch.vorburger.mariadb4j.tests.MariaDB4jSampleOtherTest
[INFO] Running ch.vorburger.mariadb4j.tests.MariaDB4jSampleDumpTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 9.15 s - in ch.vorburger.mariadb4j.tests.MariaDB4jSampleDumpTest
[INFO] Running ch.vorburger.mariadb4j.tests.junit.MariaDB4jUnitRuleInitDBTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 4.107 s - in ch.vorburger.mariadb4j.tests.junit.MariaDB4jUnitRuleInitDBTest
[INFO] Running ch.vorburger.mariadb4j.tests.junit.MariaDB4jUnitRuleOverrideDeaultPortTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 4.168 s - in ch.vorburger.mariadb4j.tests.junit.MariaDB4jUnitRuleOverrideDeaultPortTest
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   MariaDB4jSampleTutorialTest.testLocalMariaDB:57->check:70 » ManagedProcess An error occurred while installing the database
[INFO] 
[ERROR] Tests run: 19, Failures: 0, Errors: 1, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for mariaDB4j-pom 2.6.1-SNAPSHOT:
[INFO] 
[INFO] mariaDB4j-pom ...................................... SUCCESS [  5.478 s]
[INFO] mariaDB4j-core ..................................... SUCCESS [  9.334 s]
[INFO] mariaDB4j (all-in-one artifact) .................... FAILURE [01:15 min]
```

--- a/mariaDB4j/src/test/java/ch/vorburger/mariadb4j/tests/MariaDB4jSampleTutorialTest.java
+++ b/mariaDB4j/src/test/java/ch/vorburger/mariadb4j/tests/MariaDB4jSampleTutorialTest.java
@@ -2,7 +2,7 @@
  * #%L
  * MariaDB4j
  * %%
- * Copyright (C) 2012 - 2014 Michael Vorburger
+ * Copyright (C) 2012 - 2023 Michael Vorburger
  * %%
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -24,6 +24,7 @@ import static ch.vorburger.mariadb4j.DBConfiguration.Executable.Server;
 import ch.vorburger.exec.ManagedProcessException;
 import ch.vorburger.mariadb4j.DB;
 import ch.vorburger.mariadb4j.DBConfigurationBuilder;
+import java.io.File;
 import java.sql.Connection;
 import java.sql.DriverManager;
 import java.sql.SQLException;
@@ -36,7 +37,8 @@ import org.junit.Assert;
 import org.junit.Test;
 
 /**
- * Tests the functioning of MariaDB4j Sample / Tutorial illustrating how to use MariaDB4j.
+ * Tests the functioning of MariaDB4j Sample / Tutorial illustrating how to use
+ * MariaDB4j.
  *
  * @author Michael Vorburger
  * @author Michael Seaton
@@ -44,10 +46,14 @@ import org.junit.Test;
 public class MariaDB4jSampleTutorialTest {
 
     /**
-     * Tests & illustrates using MariaDB4j with an existing native MariaDB binary on the host,
+     * Tests & illustrates using MariaDB4j with an existing native MariaDB binary on
+     * the host,
      * instead of one that was bundled with and extracted from a MariaDB4j JAR.
      */
-    @Test public void testLocalMariaDB() throws Exception {
+    @Test
+    public void testLocalMariaDB() throws Exception {
+        assertExecutable("/usr/sbin/mysqld");
+
         DBConfigurationBuilder config = DBConfigurationBuilder.newBuilder();
         config.setPort(0); // 0 => autom. detect free port
         config.setUnpackingFromClasspath(false);
@@ -57,10 +63,19 @@ public class MariaDB4jSampleTutorialTest {
         check(config);
     }
 
+    private void assertExecutable(String path) {
+        if (!(new File(path).canExecute())) {
+            throw new IllegalStateException(path
+                    + " not found/executable, but required for (only) this test; try e.g. sudo apt install mariadb-server ?");
+        }
+    }
+
     /**
-     * Illustrates how to use a mysqld binary that is extracted from "embedded" binaries in JAR on classpath.
+     * Illustrates how to use a mysqld binary that is extracted from "embedded"
+     * binaries in JAR on classpath.
      */
-    @Test public void testEmbeddedMariaDB4j() throws Exception {
+    @Test
+    public void testEmbeddedMariaDB4j() throws Exception {
         DBConfigurationBuilder config = DBConfigurationBuilder.newBuilder();
         config.setPort(0); // 0 => autom. detect free port
         check(config);
@@ -106,7 +121,8 @@ public class MariaDB4jSampleTutorialTest {
         }
     }
 
-    @Test public void testEmbeddedMariaDB4jWithSecurity() throws Exception {
+    @Test
+    public void testEmbeddedMariaDB4jWithSecurity() throws Exception {
         DBConfigurationBuilder config = DBConfigurationBuilder.newBuilder();
         config.setPort(0); // 0 => autom. detect free port
         config.setSecurityDisabled(false);
@@ -147,5 +163,4 @@ public class MariaDB4jSampleTutorialTest {
             DbUtils.closeQuietly(conn);
         }
     }
-
 }

